python/zenoss/protocols/twisted/amqp.py

```diff
@@ -144,9 +148,19 @@ def get_queue(self, queue, exclusive=False):
 
     @inlineCallbacks
     def create_queue(self, queue):
+        # Check to see if the queue exists.
+        try:
+            self._checking = True
+            yield getAdapter(self.chan, IAMQPChannelAdapter).declareQueue(queue, True)
```

Comment:
Shouldn't _checking be set to false after the check? perhaps in a finally block?

Reply:
It should, yes.  I'm not at a computer for awhile to make the change but I'll get that added.